### ISSUE TITLE
Feature: support for checking list item block-id in Note Updates

### ISSUE THREAD

**Issue by MSBack** [aidenlx/obsidian-zotero#109]
Using this annot template:

```
---
callout: false
---

<% if (it.imgEmbed) { %>
- <%= it.imgEmbed %>
<% } %>
<% if (it.comment) { %>
- <% if (it.comment.startsWith('todo ')) { %>[ ] **<%= it.comment.substring(5) %>:**<% } else { %>**<%= it.comment %>:**<% } %>  
	- ==<%= it.text %>== [p. <%= it.pageLabel %>](zotero://open-pdf/library/items/<%= it.parentItem %>?page=<%= it.pageLabel %>&annotation=<%= it.key %>) ^<%= it.blockID %><% if (it.tags && it.tags.length > 0) { %> <% = it.tags.map(tag => '#' + tag.name).join(", ") %><% } %>  
<% } else if (it.text) { %>
- ==<%= it.text %>== [p. <%= it.pageLabel %>](zotero://open-pdf/library/items/<%= it.parentItem %>?page=<%= it.pageLabel %>&annotation=<%= it.key %>) ^<%= it.blockID %><% if (it.tags && it.tags.length > 0) { %> <% = it.tags.map(tag => '#' + tag.name).join(", ") %><% } %>  
<% } %>
```

Am I using the blockID wrong? Perhaps it doesn't work when comments and annotations are list items? In any case, I don't think the info from the settings is sufficient:
>If you disable callout wrapping in annotation template, you need to make sure block-id is added properly in the template.

What does *properly* mean here?

**Comment by aidenlx** [comment]
well in obsidian, block-id should be placed at the end of a block, such as regular paragraph, blockquote, callout etc. the handling of list as a block is a bit tricky. You can place it at the end of a list item to refer to specific one item, or at new line right next to the end of list to refer to the entire list. 

You can test if it works by placing block-id and check if the block can be refer to in internal link auto-complete. https://help.obsidian.md/Linking+notes+and+files/Internal+links

**Comment by MSBack** [comment]
> well in obsidian, block-id should be placed at the end of a block, such as regular paragraph, blockquote, callout etc. the handling of list as a block is a bit tricky. You can place it at the end of a list item to refer to specific one item, or at new line right next to the end of list to refer to the entire list.

I tried adding block-ids to all list items in my annot template, but all the annotations are still being reimported and duplicated (a new set of every heading and annotation), each time I update the literature note. This happens with `In-place Update of Existing Annotations` either disabled or enabled (doesn't change the update behavior).

See this screen recording:
![Obsidian_0FKDSX8hN1](https://user-images.githubusercontent.com/33608083/227908737-bf6fbcc3-d4ff-4ed2-aa19-84f6e56c84a1.gif)

At the end, I switched to Reading mode to demonstrate that the block-ids are recognized by Obsidian, since they disappear in Reading mode.

My current annot template is as follows:

```
---
callout: false
---

<% if (it.imgEmbed) { %>
- <%= it.imgEmbed %> ^<%= it.blockID %>\n
<% } %>
<% if (it.comment) { %>
- <% if (it.comment.startsWith('todo ')) { %>[ ] **<%= it.comment.substring(5) %>:**<% } else { %>**<%= it.comment %>:**<% } %> ^<%= it.blockID %>\n
	- ==<%= it.text %>== [p. <%= it.pageLabel %>](zotero://open-pdf/library/items/<%= it.parentItem %>?page=<%= it.pageLabel %>&annotation=<%= it.key %>) <% if (it.tags && it.tags.length > 0) { %> <% = it.tags.map(tag => '#' + tag.name).join(", ") %><% } %>^<%= it.blockID %>\n
<% } else if (it.text) { %>
- ==<%= it.text %>== [p. <%= it.pageLabel %>](zotero://open-pdf/library/items/<%= it.parentItem %>?page=<%= it.pageLabel %>&annotation=<%= it.key %>) <% if (it.tags && it.tags.length > 0) { %> <% = it.tags.map(tag => '#' + tag.name).join(", ") %><% } %>^<%= it.blockID %>\n
<% } %>
```

**Comment by MSBack** [comment]
Another issue here is that if I reference a block in the literature note, which already has a block-id, in another note, Obsidian will not use this block-id, but will add another one.

See:
![image](https://user-images.githubusercontent.com/33608083/227912092-4bdabe00-2dea-4659-9758-efe90540e5d6.png)

And here, embedded in another file:
![image](https://user-images.githubusercontent.com/33608083/227911302-1bf86f44-13eb-4e5c-ab5f-8a3eeb0294fa.png)

This is strange, because Obsidian supports custom block-ids like `^something` and will reuse those.

`^something` test:
![image](https://user-images.githubusercontent.com/33608083/227911572-36528bb5-b944-47bc-bf11-e8d09a2c16c5.png)

`^something` Embedded in another file:
![image](https://user-images.githubusercontent.com/33608083/227911664-48d893f9-59c0-450f-a4fb-90b3346452c9.png)

**Comment by MSBack** [comment]
I believe the output of this does have a space.

**Comment by aidenlx** [comment]
In your template all `^<%= it.blockID %>` is right next to the previous text without space in between, you can see that the block-id is not recognized properly in editor because of that.

I haven't explictly support update at the level of list item yet, so currently using this template should get updates working: 
```ejs
---
callout: false
---

<% if (it.imgEmbed) { %>
- <%= it.imgEmbed %> ^<%= it.blockID %>\n
<% } %>
<% if (it.comment) { %>
- <% if (it.comment.startsWith('todo ')) { %>[ ] **<%= it.comment.substring(5) %>:**<% } else { %>**<%= it.comment %>:**<% } %>\n
	- ==<%= it.text %>== [p. <%= it.pageLabel %>](zotero://open-pdf/library/items/<%= it.parentItem %>?page=<%= it.pageLabel %>&annotation=<%= it.key %>) <% if (it.tags && it.tags.length > 0) { %> <% = it.tags.map(tag => '#' + tag.name).join(", ") %><% } %>\n
<% } else if (it.text) { %>
- ==<%= it.text %>== [p. <%= it.pageLabel %>](zotero://open-pdf/library/items/<%= it.parentItem %>?page=<%= it.pageLabel %>&annotation=<%= it.key %>) <% if (it.tags && it.tags.length > 0) { %> <% = it.tags.map(tag => '#' + tag.name).join(", ") %><% } %>\n
<% } %>

 ^<%= it.blockID %>
 ```

**Comment by krillin666** [comment]
> okay i see the problem, I haven't handle block-id inside list items during annotation checks, I'll mark this as a feature request since I haven't explictly support update at the level of list item yet

Since I'm using this with Logseq, would it be possible to add the blockids to `property:: <%= it.blockID %>` instead of `^<%= it.blockID %>`

**Comment by aidenlx** [comment]
> since the 1.0.1 update, Obsidian now reuses the Obsidian-Zotero block-ids when I link to or embed those blocks. I am not sure what changed. Did you provide a fix for that

The update around this is not available before 1.0.2, maybe it's because your latest template fix the block-id formatting issue? The initial template introduced space after block-id, making it invaild, and the update template posted later seems to have that fixed. 

In 1.0.2, the block-id in list item are detected as well, so it should not creating duplicates now.
